File: design-patterns/python/python-adapter-design-pattern/Oracle.py
from DB import DB

class Oracle(DB):
    def __init__(self):
        self.id_counter = 0
        self.user_list = []

    def select_user_by_id(self, id):
        for user in self.user_list:
            if user.id == id:
                return user
        return None
    
    def insert_user(self, user):
        self.count()
        user.id = self.id_counter
        self.user_list.append(user)

    
    def count(self):
        self.id_counter += 1

File: design-patterns/csharp/csharp-adapter-design-pattern/Oracle.cs
using System;
using System.Collections;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace csharp_adapter
{
    public class Oracle : DB
    {
        private readonly List<User> userList;
        private int idCounter;

        public Oracle(){
            userList = new List<User>();
            idCounter = 0;
        }
        public void InsertUser(User user)
        {
            Count();
            user.Id = idCounter;
            userList.Add(user);
        }

        public void Count(){
            idCounter++;
        }

        public User SelectUserById(int id)
        {
            foreach(User user in userList){
                if(user.Id == id){
                    return user;
                }
            }
            return null;
        }
    }
}